lib/logstash/outputs/sqs.rb

```diff
@@ -1,136 +1,175 @@
 
-  # If `batch` is set to true, the number of events we queue up for a `batch_send`.
+  # The number of events to be sent in each batch. Set this to `1` to disable
+  # the batch sending of messages.
   config :batch_events, :validate => :number, :default => 10
 
-  # If `batch` is set to true, the maximum amount of time between `batch_send` commands when there are pending events to flush.
-  config :batch_timeout, :validate => :number, :default => 5
+  config :batch_timeout, :validate => :number, :deprecated => 'This setting no longer has any effect.'
 
-  public
-  def aws_service_endpoint(region)
-    return {
-        :sqs_endpoint => "sqs.#{region}.amazonaws.com"
-    }
-  end
+  # The maximum number of bytes for any message sent to SQS. Messages exceeding
+  # this size will be dropped. See
+  # http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/limits-messages.html.
+  config :message_max_size, :validate => :bytes, :default => '256KiB'
 
-  public 
+  # The name of the target SQS queue. Note that this is just the name of the
+  # queue, not the URL or ARN.
+  config :queue, :validate => :string, :required => true
+
+  public
   def register
-    require "aws-sdk"
-
-    @sqs = AWS::SQS.new(aws_options_hash)
-
-    if @batch
-      if @batch_events > 10
-        raise RuntimeError.new(
-          "AWS only allows a batch_events parameter of 10 or less"
-        )
-      elsif @batch_events <= 1
-        raise RuntimeError.new(
-          "batch_events parameter must be greater than 1 (or its not a batch)"
-        )
-      end
-      buffer_initialize(
-        :max_items => @batch_events,
-        :max_interval => @batch_timeout,
-        :logger => @logger
-      )
-    end
+    @sqs = Aws::SQS::Client.new(aws_options_hash)
+
+    if @batch_events > 10
+      raise LogStash::ConfigurationError, 'The maximum batch size is 10 events'
+    elsif @batch_events < 1
+      raise LogStash::ConfigurationError, 'The batch size must be greater than 0'
+   end
 
     begin
-      @logger.debug("Connecting to AWS SQS queue '#{@queue}'...")
-      @sqs_queue = @sqs.queues.named(@queue)
-      @logger.info("Connected to AWS SQS queue '#{@queue}' successfully.")
-    rescue Exception => e
-      @logger.error("Unable to access SQS queue '#{@queue}': #{e.to_s}")
-    end # begin/rescue
-  end # def register
+      @logger.debug('Connecting to SQS queue', :queue => @queue, :region => region)
+      @queue_url = @sqs.get_queue_url(:queue_name => @queue)[:queue_url]
+      @logger.info('Connected to SQS queue successfully', :queue => @queue, :region => region)
+    rescue Aws::SQS::Errors::ServiceError => e
+      @logger.error('Failed to connect to SQS', :error => e)
+      raise LogStash::ConfigurationError, 'Verify the SQS queue name and your credentials'
+    end
+  end
 
   public
-  def receive(event)
-    if @batch
-      buffer_receive(event.to_json)
-      return
+  def multi_receive_encoded(encoded_events)
```

Comment:
See this gist for an implementation of what I suggested below: https://gist.github.com/jrgns/e4fc656582a624bfd7b8040a3260a648